go/arrow/csv/writer.go

```diff
@@ -219,6 +219,38 @@ func (w *Writer) Write(record arrow.Record) error {
 					recs[i][j] = w.nullValue
 				}
 			}
+		case *arrow.Decimal128Type:
+			fieldType := w.schema.Field(j).Type.(*arrow.Decimal128Type)
+			scale := fieldType.Scale
+			arr := col.(*array.Decimal128)
+			for i := 0; i < arr.Len(); i++ {
+				if arr.IsValid(i) {
+					decString := arr.Value(i).BigInt().Text(10)
+					decimalLocation := len(decString) - int(scale)
+					if decimalLocation >= 0 && scale > 0 {
+						decString = decString[:decimalLocation] + "." + decString[decimalLocation:]
+					}
+					recs[i][j] = decString
```

Comment:
Any particular reason not to follow the setup used for marshalling to json? 

```go
f := (&big.Float{}).SetInt(arr.Value(i).BigInt())
f.Quo(f, big.NewFloat(math.Pow10(int(scale))))
recs[i][j] = f.Text('g', int(precision))
```
The same thing works for decimal256

Reply:
None other than I failed to think to check what is done for JSON :)

This is much cleaner, thanks!

Reply:
No prob! At this rate I should probably just incorporate this to the `decimal128.Num` and `decimal256.Num` types with a `ToString(precision, scale int32) string` method, but that can be a separate change. :smile: